dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/McpRequestIdDeserializer.java

```
@@ -26,24 +27,27 @@
 /**
  * Instructions for how Jsonb should deserialize JSON values into a {@link McpRequestId} type
  */
-public class McpRequestIdDeserializer implements JsonbDeserializer<McpRequestId> {
+public class McpRequestIdDeserializer implements JsonbDeserializer<RequestId> {
     private static final TraceComponent tc = Tr.register(McpRequestIdDeserializer.class);
 
     @Override
-    public McpRequestId deserialize(JsonParser parser, DeserializationContext ctx, Type rtType) {
+    public RequestId deserialize(JsonParser parser, DeserializationContext ctx, Type rtType) {
+
         JsonValue jsonVal = parser.getValue();
         switch (jsonVal.getValueType()) {
             case STRING:
                 String strVal = ((JsonString) jsonVal).getString();
                 if (strVal.isBlank())
-                    throw new JSONRPCException(JSONRPCErrorCode.PARSE_ERROR, Tr.formatMessage(tc, "CWMCM0019E.jsonrpc.validation.empty.string.id", strVal));
-                return new McpRequestId(strVal);
+                    throw new JSONRPCException(JSONRPCErrorCode.PARSE_ERROR,
+                                               Tr.formatMessage(tc, "CWMCM0019E.jsonrpc.validation.empty.string.id", strVal));
+                return new RequestId(strVal);
             case NUMBER:
-                return new McpRequestId(((JsonNumber) jsonVal).bigDecimalValue());
+                return new RequestId(((JsonNumber) jsonVal).bigDecimalValue());
+            case NULL:
+                return new RequestId("Request ID must not be null");
```

I think this case will still create the RequestId, but set the id value to `"Request ID must not be null"`, so can probably just remove the `NULL` case and let it throw the default error